app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt

```diff
@@ -0,0 +1,125 @@
+package com.tomtom.sdk.examples.startnavigation
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import com.tomtom.sdk.examples.BuildConfig
+import com.tomtom.sdk.location.GeoLocation
+import com.tomtom.sdk.location.GeoPoint
+import com.tomtom.sdk.location.LocationProvider
+import com.tomtom.sdk.location.simulation.SimulationLocationProvider
+import com.tomtom.sdk.location.simulation.strategy.InterpolationStrategy
+import com.tomtom.sdk.navigation.NavigationConfiguration
+import com.tomtom.sdk.navigation.NavigationOptions
+import com.tomtom.sdk.navigation.ProgressUpdatedListener
+import com.tomtom.sdk.navigation.RoutePlan
+import com.tomtom.sdk.navigation.TomTomNavigation
+import com.tomtom.sdk.navigation.TomTomNavigationFactory
+import com.tomtom.sdk.navigation.routereplanner.RouteReplanner
+import com.tomtom.sdk.navigation.routereplanner.online.OnlineRouteReplannerFactory
+import com.tomtom.sdk.routing.RoutePlanner
+import com.tomtom.sdk.routing.RoutePlanningCallback
+import com.tomtom.sdk.routing.RoutePlanningResponse
+import com.tomtom.sdk.routing.RoutingFailure
+import com.tomtom.sdk.routing.online.OnlineRoutePlanner
+import com.tomtom.sdk.routing.options.Itinerary
+import com.tomtom.sdk.routing.options.RoutePlanningOptions
+import com.tomtom.sdk.routing.options.guidance.*
+import com.tomtom.sdk.routing.route.Route
+import com.tomtom.sdk.vehicle.Vehicle
+
+class NavigationActivity : AppCompatActivity() {
+    private lateinit var tomTomNavigation: TomTomNavigation
+
+    private lateinit var locationProvider: LocationProvider
+    private lateinit var routePlanner: RoutePlanner
+    private lateinit var routeReplanner: RouteReplanner
+    private var route: Route? = null
+    private var routePlanningOptions: RoutePlanningOptions? = null
+
+    private val apiKey = BuildConfig.TOMTOM_API_KEY
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initRouting()
+        planRoute(amsterdam, rotterdam)
+    }
+
+    private fun initRouting() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = apiKey)
+        routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
```

Comment:
do we need `routeReplanner`?

Reply:
But in retrospect, maybe we should just pass the route planner here. What is a route replanner, by the way?